src/libtiled/maptovariantconverter.cpp

```diff
@@ -396,6 +418,7 @@ QVariant MapToVariantConverter::toVariant(const WangColor &wangColor) const
     colorVariant[QStringLiteral("probability")] = wangColor.probability();
     colorVariant[QStringLiteral("tile")] = wangColor.imageId();
     addProperties(colorVariant, wangColor.properties());
+    addComponents(colorVariant, wangColor.components());
```

Comment:
Maybe we should just change `addProperties` to `addPropertiesAndComponents(QVariantMap &variantMap, const Object &object)`?

Reply:
I suggest against `addPropertiesAndComponents` or alike. It would be a type of function doing two inherently different things. That's why I put read/write components calls next to read/write/etc properties calls all throughout the code.